src/components/UserList.tsx

```diff
@@ -13,18 +20,43 @@ class User {
   }
 }
 
-interface UserListProp {
-  users: Array<User>;
-}
+const data = {
+  users: [
+    { id: '1', name: 'admin', role: 1 },
+    { id: '2', name: 'normal', role: 0 },
+  ],
+};
+
+const UserList: FC<RouteComponentProps> = ({ match }) => {
```

Comment:
this code is for nested routing.
https://reactrouter.com/web/guides/quick-start/2nd-example-nested-routing

nested routing will be replaced by **nested routes** feature.